### PR TITLE
Refactor the react index.gsp files

### PR DESCRIPTION
**Pull request by rushtong**
## Changes
* Make all of the primary react-based `index.gsp` files use the same template. 
* Pull all constants into the same object(s) so they can be reused.

## Testing
View search, project, consent group pages - should be no errors

## TODO
There is a lot we can do to refactor further, but I leave that for other PRs. This PR is meant to start us in the right direction in terms of consolidating react behavior. The more we can pull out of individual gsps, the easier it will be to consolidate them later on as well as convert other features more easily.

Also - this does not address the templates we are using (`_index.gsp` files). We need to figure that out too. 

---

- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Get a thumb from @rushtong
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

File: grails-app/views/layouts/main_component.gsp [hunk 1]
```diff
@@ -0,0 +1,184 @@
+<%@ page import="org.broadinstitute.orsp.IssueStatus" %>
+<%@ page import="org.broadinstitute.orsp.IssueType" %>
+<%@ page import="org.broadinstitute.orsp.PreferredIrb" %>
+
+<!DOCTYPE html>
```
Comment (rushtong) [inline grails-app/views/layouts/main_component.gsp]: This file is a straight copy of `main.gsp` with additions to support the react stuff. I envision this being where we start doing the consolidation and eventually deprecate `main.gsp` altogether.

Comment (tadeork) [reply grails-app/views/layouts/main_component.gsp]: So in the future we'll use this as a main entry-point to all the info that the components will need?

Comment (rushtong) [reply grails-app/views/layouts/main_component.gsp]: That is the idea, yes. We could instead use the old `main.gsp` but I wanted to try and minimize the introduction of bugs with the old code.

**Comment by rushtong** [comment]
Thanks @tadeork - will merge and if we see problems, will address immediately.

**Pull request merged**